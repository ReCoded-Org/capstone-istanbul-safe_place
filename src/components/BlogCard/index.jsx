import React from "react";
import { Col, Card } from "react-bootstrap";
import "./index.scss";

const BlogCard = ({ blogPost }) => {
  return (
    <Col md={4}>
      <Card className="blogPost">
        <Card.Img variant="top" src={blogPost.jetpack_featured_media_url} />
        <Card.Body>
          <Card.Title
            className="blogPostTitle"
            dangerouslySetInnerHTML={{
              __html: blogPost.title.rendered,
            }}
          />
        </Card.Body>
      </Card>
    </Col>
  );
};

export default BlogCard;
