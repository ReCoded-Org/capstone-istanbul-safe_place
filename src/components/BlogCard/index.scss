@import "../../style/variables";

.blogPost {
  height: 20rem;
  border-radius: 1rem !important;
  box-shadow: 0px 0px 10px 0 rgba(0, 0, 0, 0.06);
  margin-bottom: 2rem;

  img {
    border-top-left-radius: calc(1rem - 1px) !important;
    border-top-right-radius: calc(1rem - 1px) !important;
    object-fit: cover !important;
    object-position: 0 25% !important;
    height: 15rem;
  }

  .blogPostTitle {
    text-align: center;
    color: $primaryColor;
  }

  &:hover {
    text-decoration: underline;
    color: $primaryColor;
  }

  a {
    color: $primaryColor;
  }
}
