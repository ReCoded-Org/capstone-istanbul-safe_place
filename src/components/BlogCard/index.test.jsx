import React from "react";
import { render , unmountComponentAtNode} from "react-dom";
import { act } from "react-dom/test-utils";
import BlogCard from "./index";


let container;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  document.body.removeChild(container);
  container = null;
});

describe("BlogCard", () => {
  const MOCK_DATA = [
    {
      title: {
        rendered: "You can find safe place for yourself here",
      },
      jetpack_featured_media_url:
        "https://safeplace102505649.files.wordpress.com/2020/11/womentogether.jpg",
      id: 78,
    },
  ];

  it("Renders without crashing", async () => {
    jest.spyOn(global, "fetch").mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(MOCK_DATA),
      })
    );

    await act(async () => {
      render(<BlogCard />, container);
    });

    expect(container).toMatchSnapshot();
  });
});
