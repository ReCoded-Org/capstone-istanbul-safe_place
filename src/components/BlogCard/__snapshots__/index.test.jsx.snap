// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<div
  className="col-md-4"
>
  <a
    href="/blog/78"
    onClick={[Function]}
  >
    <div
      className="blogPost card"
    >
      <img
        className="card-img-top"
        src="https://safeplace102505649.files.wordpress.com/2020/11/womentogether.jpg"
      />
      <div
        className="card-body"
      >
        <div
          className="blogPostTitle card-title h5"
          dangerouslySetInnerHTML={
            Object {
              "__html": "You can find safe place for yourself here",
            }
          }
        />
      </div>
    </div>
  </a>
</div>
`;
